test/index.html

```diff
@@ -0,0 +1,247 @@
+        // However, other QUnit.test() blocks than this will fail miserably if it
+        // weren't supported.
+        assert.ok( !document.body.classList.contains("impress-disabled"),
+                   "body.impress-disabled is removed." );
+        assert.ok( document.body.classList.contains("impress-enabled"),
+                   "body.impress-enabled is added." );
+        
+        var canvas = document.querySelector( "div#impress > div" );
+        assert.ok( !canvas.classList.contains("step") && canvas.id === "",
+                   "Additional 'canvas' div inserted between div#impress root and steps." );
+        assert.equal( canvas.style.transform,
+                      "rotateZ(0deg) rotateY(0deg) rotateX(0deg) translate3d(1000px, 0px, 0px)",
+                      "canvas.style.transform initialized correctly" );
+        assert.equal( canvas.style.transformOrigin,
+                      "left top 0px",
+                      "canvas.style.transformOrigin initialized correctly" );
+        assert.equal( canvas.style.transformStyle,
+                      "preserve-3d",
+                      "canvas.style.transformStyle initialized correctly" );
+        assert.equal( canvas.style.transitionDelay,
+                      "0ms",
+                      "canvas.style.transitionDelay initialized correctly" );
+        // impress.js default values tries to set this to 1000ms, I'm completely confused about why that's not actually set in the browser?
+        assert.equal( canvas.style.transitionDuration,
+                      "0ms",
+                      "canvas.style.transitionDuration initialized correctly" );
+        assert.equal( canvas.style.transitionProperty,
+                      "all",
+                      "canvas.style.transitionProperty initialized correctly" );
+        assert.equal( canvas.style.transitionTimingFunction,
+                      "ease-in-out",
+                      "canvas.style.transitionTimingFunction initialized correctly" );
+                      
+        assert.equal( document.documentElement.style.height,
+                      "100%",
+                      "documentElement.style.height is 100%" );
+        
+        // Steps initialization
+        var step1 = document.querySelector( "div#step-1" );
+        assert.equal( step1.style.position,
+                      "absolute",
+                      "Step position is 'absolute'." );
+
+        assert.ok( step1.classList.contains("active"),
+                   "Step 1 has active css class." );
+      }
+    });
+
+    // Note: Here we focus on testing the core functionality of moving between
+    // steps, the css classes set and unset, and events triggered.
+    // TODO: more complex animations and check position, transitions, delays, etc...
+    // Those need to be separate html files, and there could be several of them.
+    QUnit.test( "Impress Core API", function( assert ) {
```

Comment:
> Are you referring to the TODO line?

I was referring more to the fact that "Impress Core API" is declared as a `QUnit.test` instead of a `QUnit.module`. The name "Impress Core API" represents a set of tests instead of a single one. My suggestion would be something like this:

```javascript
QUnit.module( "Impress Core API" );
...
```

> This is because tests operate on the div.step elements we've defined directly under body. So we can only have one impress.js presentation in one html file.

Ahh, now I see. Having one test seems to be more a required need due to how impress works than an actual decision. In this case, to organise the tests accordingly inside each `test`, wouldn't be better run each one in a dedicated iframe that loads an external HTML file for each time the presentation should start from scratch? That would remove the worries about impress creating global data in the body element (or at least until we do it in a way where you can start more than one impress instance in the same page).